### ISSUE TITLE
Running CLF on basic sklearn datasets

### ISSUE THREAD

**Issue by bscully** [scikit-learn-contrib/boruta_py#24]
Python2.7 64, sklearn .18.1, Boruta .1.5

Results of running CLF on basic sklearn datasets:
IRIS fails
BREAST_CANCER fails
DIGITS completes

There error is:
  File "C:\Python27\lib\site-packages\scipy\stats\mstats_basic.py", line 254, in _rank1d
    for r in repeats[0]:
TypeError: iteration over a 0-d array

RandomForestClassifier params: n_estimators=3, max_depth=3
BorutaPy params: perc=100, alpha=.01

For BREAST_CANCER, when changing the classifier to max_depth of 1, the code runs.
For IRIS, when changing the classifier to max_depth of 1, the code still fails.  Further, it seems like no parameters work for the IRIS dataset.

Is there a way to improve the stability?  I saw this old issue/commit but the error seems more extensive.
https://github.com/scikit-learn-contrib/boruta_py/commit/80a74c1989b05c893210cb04ac754498723d1c97

**Comment by danielhomola** [comment]
Hi there,

Thanks for reporting this. (Next time could you please include the part of the traceback that's coming from boruta_py as well?)

This is an interesting one and took me a while to figure out. Basically, Boruta fails here because all features are selected from the dataset. Since I was developing and benchmarking on Boruta on real datasets not toy data, I never ran into a scenario like this. 

I'll add a safety check to prevent this from happening.

**Comment by bscully** [comment]
excellent, thank you for addressing.

**Comment by farfan92** [comment]
Has this been addressed yet? I seem to be running into the same issue

**Comment by danielhomola** [comment]
no.. I'm really sorry.. I was busy writing up my thesis.. my viva is on Friday, should be more productive in the coming weeks..
